post/views.py

```diff
@@ -1,8 +1,12 @@
 from django.shortcuts import render,HttpResponse
 from django.views.generic import ListView, CreateView, DetailView
-from django.urls import reverse 
+from django.urls import reverse_lazy 
 from .forms import PostForm
 from .models import Post
+from django.contrib.auth.models import User
+from post.serializers import UserSerializer
+from rest_framework import generics
+from rest_framework import permissions
```

Comment:
When importing multiple things from one source, it's generally better to combine them.

```suggestion
from rest_framework import generics, permissions
```